internal/etcd/etcd.go

```diff
@@ -149,6 +151,78 @@ func (c *ClientEtcdAPI) Close() error {
 	return nil
 }
 
+// UsedSpacePercentage returns the used space as a percentage of the storage quota
+func (c *ClientEtcdAPI) UsedSpacePercentage(ctx context.Context, member Member) (int, error) {
+	if len(member.ClientURLs) < 1 {
+		return 0, fmt.Errorf("unable to get used space, etcd member has no client urls")
+	}
+	url := member.ClientURLs[0] + "/metrics"
+	resp, err := http.Get(url)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get metrics from etcd: %w", err)
+	}
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return 0, fmt.Errorf("failed to read metrics returned from etcd: %w", err)
+	}
+	defer resp.Body.Close()
```

Comment:
I feel like this could be a separate `GetMetrics` function?

Reply:
Yep, I've separate this now